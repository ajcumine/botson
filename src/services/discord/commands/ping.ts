import { Command } from './';

export const execute = message => {
  message.channel.send('Pong.');
};

const ping: Command = {
  execute,
  argsRequired: false,
  description: 'Ping!',
  guildOnly: false,
};

export default ping;
