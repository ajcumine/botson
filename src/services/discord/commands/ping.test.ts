import ping, { execute } from './ping';

describe('!ping command', () => {
  it('handles !ping commands', () => {
    expect(ping).toMatchSnapshot();
  });

  describe('execute', () => {
    it('responds with "Pong."', () => {
      const messageSendSpy = jest.fn();
      const mockMessage = {
        channel: {
          send: messageSendSpy,
        },
      };

      execute(mockMessage);
      expect(messageSendSpy).toHaveBeenCalledWith('Pong.');
    });
  });
});
