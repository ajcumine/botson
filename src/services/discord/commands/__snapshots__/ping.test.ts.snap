// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!ping command handles !ping commands 1`] = `
Object {
  "argsRequired": false,
  "description": "Ping!",
  "execute": [Function],
  "guildOnly": false,
}
`;
